### PR TITLE
CRM-20308 added method getReceiptFrom to set FROM in sendActivityCopy…

### PR DESCRIPTION
**Pull request by ErikHommel**
… - send email to assignee

### CODE REVIEW

**Comment by seamuslee001** [comment]
@ErikHommel can you change the permissions on the file back to 644 

I think this makes sense to be but don't know CiviCase enough pinging @colemanw

File: CRM/Case/BAO/Case.php [hunk 1]
```diff
@@ -3168,4 +3167,58 @@ public function addSelectWhereClause() {
...
+    $emailFromContactId = NULL;
+    if (!empty($activityId)) {
+      try {
+        $sourceRecordTypeId = civicrm_api3('OptionValue', 'getvalue', array(
+          'option_group_id' => 'activity_contacts',
+          'name' => 'Activity Source',
+          'return' => 'value'));
+        $emailFromContactId = civicrm_api3('ActivityContact', 'getvalue', array(
+          'activity_id' => $activityId,
+          'record_type_id' => $sourceRecordTypeId,
```
Comment (eileenmcnaughton) [inline CRM/Case/BAO/Case.php]: The api will resolve this (with better caching) if you do 

'record_type_id' => 'Activity Source'

File: CRM/Case/BAO/Case.php [hunk 2]
```diff
@@ -3168,4 +3167,58 @@ public function addSelectWhereClause() {
...
+          'return' => 'contact_id'
+        ));
+      } catch (CiviCRM_API3_Exception $ex) {
+        // get default from address from domain
+        $domainId = CRM_Core_Config::domainID();
```
Comment (eileenmcnaughton) [inline CRM/Case/BAO/Case.php]: @seamuslee001 didn't you write a function that does this stuff?

Comment (seamuslee001) [inline CRM/Case/BAO/Case.php]: I did but that was bundled up with receipting for contributions but yeh might be interesting if we can combine the two

**Comment by totten** [comment]
jenkins, add to whitelist

**Comment by eileenmcnaughton** [comment]
@ErikHommel jenkins hit some style warnings

**Comment by seamuslee001** [comment]
This is the function you were talking about Eileen https://github.com/civicrm/civicrm-core/blob/master/CRM/Contribute/BAO/Contribution.php#L4758

**Comment by eileenmcnaughton** [comment]
Ah yeah - maybe extracting from here down into a function https://github.com/civicrm/civicrm-core/blob/master/CRM/Contribute/BAO/Contribution.php#L4773 would create a chunk that can be used for this purpose

**Comment by ErikHommel** [comment]
Updated with QA suggestions, thanks!

**Comment by eileenmcnaughton** [comment]
You still haven't impressed jenkins :-) - ping me if you have trouble understanding how to get onside with jenkinds

File: CRM/Case/BAO/Case.php [hunk 3]
```diff
@@ -3168,4 +3167,59 @@ public function addSelectWhereClause() {
...
+   * @see https://issues.civicrm.org/jira/browse/CRM-20308
+   */
+  private static function getReceiptFrom($activityId) {
+    $emailFromContactId = NULL;
+    if (!empty($activityId)) {
+      try {
+        $emailFromContactId = civicrm_api3('ActivityContact', 'getvalue', array(
+          'activity_id' => $activityId,
+          'record_type_id' => 'Activity Source',
+          'return' => 'contact_id',
```
Comment (seamuslee001) [inline CRM/Case/BAO/Case.php]: don't you need to return email here

File: CRM/Case/BAO/Case.php [hunk 4]
```diff
@@ -3168,4 +3167,59 @@ public function addSelectWhereClause() {
...
+        ));
+      }
+      catch (CiviCRM_API3_Exception $ex) {
+        // get default from address from domain
+        try {
```
Comment (seamuslee001) [inline CRM/Case/BAO/Case.php]: From here down i would recommend replacing with the following 

```
list($name, $address) = CRM_Core_BAO_Domain::getDefaultReceiptFrom();
then return "$name <$address>"

File: CRM/Case/BAO/Case.php (hunk 3 above)
Comment (seamuslee001) [inline CRM/Case/BAO/Case.php]: You should return display_name and email here IMO and then do 

```
return $emailFromContactId['display_name'] . '<' . $emailFromContactId['email'] . '>';
```

**Comment by monishdeb** [comment]
@ErikHommel I have cherry-picked your commit and submitted a new PR #10158 with some additional changes and added a unit test. Please have a look.

**Comment by monishdeb** [comment]
Closing this PR in favor of #10158

**Pull request closed without merging**